incubator/orm/indexer.go

```diff
@@ -1,135 +1,137 @@
 package orm
 
 import (
-	"encoding/binary"
-	"math"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-// IndexerFunc creates one or multiple MultiKeyIndex keys for the source object.
-type IndexerFunc func(value interface{}) ([][]byte, error)
+// IndexerFunc creates one or multiple index keys for the source object.
+type IndexerFunc func(value interface{}) ([]RowID, error)
 
 // IndexerFunc creates exactly one index key for the source object.
-type UniqueIndexerFunc func(value interface{}) ([]byte, error)
+type UniqueIndexerFunc func(value interface{}) (RowID, error)
 
-// Indexer manages the persistence for an MultiKeyIndex based on searchable keys and operations.
+// Indexer manages the persistence for an Index based on searchable keys and operations.
 type Indexer struct {
-	indexerFunc IndexerFunc
-	addPolicy   func(store sdk.KVStore, secondaryIndexKey []byte, rowID uint64) error
+	indexerFunc   IndexerFunc
+	addPolicy     func(store sdk.KVStore, codec IndexKeyCodec, secondaryIndexKey []byte, rowID RowID) error
```

Comment:
Maybe something like `addFunction` would be more intuitive. Policy sounds like just a validation function, but this actually mutates state.